torch/csrc/utils/future.h

```diff
@@ -109,41 +114,40 @@ class TORCH_API Future final {
   }
 
   // If completed() the callback will be invoked in-place.
-  void addCallback(const Callback& callback) {
+  void addCallback(std::function<void(void)> cb) {
     std::unique_lock<std::mutex> lock(mutex_);
     if (completed_) {
       lock.unlock();
-      callback(value_, error_);
-      return;
```

Comment:
@jjlilley Do we really want to remove this return?

Reply:
Good catch, thank you!

The good news is the impact should be relatively minimal (it would exacerbate memory leaks case by reference cycles), but definitely a bug.

Fix at: https://github.com/pytorch/pytorch/pull/36662